app/models/user.rb

```diff
@@ -17,6 +17,8 @@ class User < ApplicationRecord
   has_many :companies, through: :billing_contracts
   has_many :payments, through: :billing_contracts
   has_many :news, dependent: :nullify
+  has_many :neighbors, ->(user) { where('osbb_id = ? and users.id != ?', user.osbb_id, user.id) },
```

Comment:
no need for `osbb_id = ?` if you already have `through: :osbb `

Reply:
create `scope non_self` or smth like that 
to make this check `users.id != ?`

Reply:
done